js/data.js

```diff
@@ -1,8 +1,12 @@
 /* exported data */
 
-var data = {
-  view: 'entry-form',
-  entries: [],
-  editing: null,
-  nextEntryId: 1
-};
+if (localStorage.getItem('data')) {
+  var data = JSON.parse(localStorage.getItem('data'));
+} else {
+  data = {
+    view: 'entry-form',
+    entries: [],
+    editing: null,
+    nextEntryId: 1
+  };
```

Comment:
I would keep the variable declaration here as it was so that there is an easy place to go to see the structure of the data. With the `var` in a separate block here, this is potentially confusing. You can have the conditional to check the value in `localStorage` below the variable declaration.